python/ray/autoscaler/_private/autoscaler.py

```diff
@@ -388,6 +399,43 @@ def terminate_scheduled_nodes(self):
         self.nodes_to_terminate = []
         self.update_worker_list()
 
+    def drain_nodes_via_gcs(self):
+        node_ips = [
+            self.provider.internal_ip(node_id)
+            for node_id in self.nodes_to_terminate
+        ]
+        raylet_ids_to_drain = {
```

Comment:
Do we need to get raylet id here? I think node id == raylet id?

Reply:
The confusing thing is that the node_id two lines up (and in most other places in the autoscaler) is a NodeProvider-provided node id independent of the ray-internal node id.

The logic here converts (node provider node id) -> ip -> (ray-internal node id)

Will add some comments and re-name some variables.

Reply:
comments/clearer names would be great here. Thanks!

Reply:
Added some comments and, used the term "provider node id" for the relevant variables.